my_enumerable.rb

```diff
@@ -0,0 +1,154 @@
+
+# rubocop:disable all
+module Enumerable
+  def my_each
+    for item in self
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    for item in self
+      yield(item, index(item))
+    end
+  end
+
+  def my_select
+    for element in self
+      yield(element)
+   end
+ end
+
+  def my_any?(args = nil)
+    result = true
+    if !block_given? && args.nil?
+      empty? ? result = false : nil
+      my_each do |item|
+        if is_a? Hash
+          result = false if item[1] == false || item[1].nil?
+        elsif item.nil?
+          result = false
+        else
+          result = false if item == false || item.nil?
+        end
+      end
+      result
+    else
+      my_select { |key, value| yield(key, value) } if is_a? Hash
+      my_each { |key| yield(key) }
+    end
+  end
```

Comment:
- [ ] `my_any?`:
- returns true if the block ever returns a value other than false or nil
`block = proc { |num| num <= 3 }`
`p [12,12,13,1,4].my_any?(&block)` should return `true`

- when no block or argument is given returns true if at least one of the collection is not false or nil
`p [nil,true, nil].my_any?` should return `true`
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
`p [12,3,34].my_any?(Integer)` should return `true`
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
`p ['ghetto','wild'].my_any?(/z/)` should return `false`
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
`p ['cat',2,32,].my_any?('cat')` should return `true`